packages/adapter-mysql/src/index.ts

```diff
@@ -0,0 +1,273 @@
+    },
+    async createSession({ sessionToken, userId, expires }) {
+      if (userId === undefined) {
+        throw Error(`userId is undef in createSession`);
+      }
+      const sql = `insert into sessions (userId, expires, sessionToken)
+      values (?, ?, ?)
+    `;
+
+      const [result] = (await client.query(sql, [
+        userId,
+        expires,
+        sessionToken,
+      ])) as OkPacketParams[];
+      if (result.insertId)
+        return { id: result.insertId, sessionToken, userId, expires };
+      else throw new Error("nothing created ");
+    },
+
+    async getSessionAndUser(sessionToken: string | undefined): Promise<{
+      session: AdapterSession;
+      user: AdapterUser;
+    } | null> {
+      if (sessionToken === undefined) {
+        return null;
+      }
+      const [result1] = await client.query(
+        `select * from sessions where sessionToken = ?`,
+        [sessionToken]
+      );
+      const [oldUser] = result1 as AdapterSession[];
+
+      if (!oldUser) {
+        return null;
+      }
+
+      const [result2] = await client.query("select * from users where id = ?", [
+        oldUser.userId,
+      ]);
+      const [newUser] = result2 as AdapterUser[];
+
+      if (!newUser) {
+        return null;
+      }
+      return {
+        session: oldUser,
+        user: newUser,
+      };
+    },
+    async updateSession(
+      session: Partial<AdapterSession> & Pick<AdapterSession, "sessionToken">
+    ): Promise<AdapterSession | null | undefined> {
+      const { sessionToken } = session;
+      const [result1] = await client.query(
+        `select * from sessions where sessionToken = ?`,
+        [sessionToken]
+      );
+      const [originalSession] = result1 as AdapterSession[];
+
+      if (!originalSession) {
+        return null;
+      }
+
+      const newSession: AdapterSession = {
+        ...originalSession,
+        ...session,
+      };
+      const sql = `
+        UPDATE sessions set
+        expires = ?
+        where sessionToken = ?
+        `;
+      const [result] = (await client.query(sql, [
+        newSession.expires,
+        newSession.sessionToken,
+      ])) as OkPacketParams[];
+      return result.affectedRows ? newSession : null;
+    },
+    async deleteSession(sessionToken) {
+      const sql = `delete from sessions where sessionToken = ?`;
+      await client.query(sql, [sessionToken]);
+    },
+    async unlinkAccount(partialAccount) {
+      const { provider, providerAccountId } = partialAccount;
+      const sql = `delete from accounts where providerAccountId = ? and provider = ?`;
+      await client.query(sql, [providerAccountId, provider]);
+    },
+    async deleteUser(userId: string) {
+      await client.query(`delete from users where id = ?`, [userId]);
+      await client.query(`delete from sessions where userId = ?`, [userId]);
+      await client.query(`delete from accounts where userId = ?`, [userId]);
+      return null;
+    },
```

Comment:
this changes should be wrapped in transaction

```suggestion
    async deleteUser(userId: string) {
      await client.beginTransaction();
      try {
        await client.query(`delete from ${userTable} where id = ?`, [userId]);
        await client.query(`delete from sessions where userId = ?`, [userId]);
        await client.query(`delete from accounts where userId = ?`, [userId]);
        await client.commit();
      } catch (error) {
        await client.rollback();
        throw error;
      }
      return null;
    },
```